_posts/2022-02-23-sparse-models.md

```diff
@@ -54,107 +61,35 @@ LTH and successive works have shown that IMP and variants (such as Layer-adaptiv
-In the figures below, the architectures are ordered in increasing order of the number of parameters.
-
-<div align="center">
-{% include 2022-02-07-sparse-models/difficulty.html %}
-</div>
-
-For the architectures they tested, they conclude that, in general, it is `easier` to find winning tickets for smaller
-architectures; which is a contradictory to the assumption of LTH.
-
-### Quality of Winning Tickets
-
-Since larger architectures have more number of sub-networks, one would except to see the highest accuracy
-gain as compared to smaller architectures, as there is a higher chances of better-quality
-sub-networks to exist in larger architectures.
-
-
-<div align="center">
-{% include 2022-02-07-sparse-models/quality.html %}
-</div>
-
-Again, it is observed that ResNet18, which is the larget architecture has the lowest accuracy gain when
-compared to its dense counterpart.
+## Can we find Winning Tickets from Pre-trained models?
+
+One of the statements of LTH is that the network needs to be rewinded to its initial state. However, it is a common use-case that we take a large pre-trained model such as Wav2Vec2 and then finetune on a down-stream task which usually is a much smaller dataset.
+
+Can we treat these pre-trained weights as the "initial state" in LTH and be able to find sparse models? Authors from [3] show that this is true. They show that they were able to prune away upto 95% of weights while achieving comparable WER on multiple architectures, ranging from CNN-LSTM, RNN-Transducer to Conformer and two public datasets (Common Voice and LibriSpeech).
+
+<figure>
+<center>
+  <img alt="Ticket Search for Pretrained Models" src="/assets/images/posts/sparse-models/pretrained.jpeg"/>
+  <figcaption>Fig 2: Iterative Pruning for Pretrained Models. x-axis: Percentage of Weights remaining after each pruning stage. y-axis: Word Error Rate (WER) % </figcaption>
+</center>
+</figure>
+
+In Fig 2, $$\theta_0$$ denotes random initialization, $$\theta_{CV}$$ denotes transfering from weights trained on CommonVoice, and $$\theta_{Libri}$$ denotes transfering from weights trained on LibriSpeech. WER is reported on TED-LIUM dataset. 
+
+We can observe that, as expected, models initialized with pretrained weights have a lower WER, however, the drop in WER due to Ticket Search is lower but comparable to non-pretrained initialization.
 
 
 ## Conclusion
+Pruning, traditionally only thought of as a way to speedup inference. However, The Lottery Ticket Hypothesis shows that iterative pruning can improve performance and robustness of models, in-addition to speeding up inference!
 
-As a last-mile effort in your Machine Learning project, you should consider Iterative Pruning methods to improve generalizability, latency and memory requirements, even if your architecture is already very small/lightweight!
+Iterative Pruning (or Ticket Search) is a tool that should be considered to be used at least as a last-ditch effort for your production models.
```

Comment:
How does this compare to other 'speeding' techniques? When we are arguing that people should be using pruning, we need to also tell them the sibling concepts. Both around other hypothesis around why pruning works, and why pruning works better (similar, worse) than other methods.

Reply:
Most popular inference optimization methods are orthogonal. You can prune and quantize and fuse operations all at the same time. Quantization and Pruning or Pruning vs Layer Fusion is an apples-orange comparison. In fact, most optimizations are done at the same time in most production settings (at least in my experience).